File: 2024/day-3/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func ArrayAtoi(t []string) []int {
	var output []int
	for _, value := range t {
		y, err := strconv.Atoi(value)
		check(err)
		output = append(output, y)
	}
	return output
}

func main() {
	file, err := os.Open("input.txt")
	check(err)
	fileScanner := bufio.NewScanner(file)
	fileScanner.Split(bufio.ScanLines)

	total := 0
	for fileScanner.Scan() {
		lineStr := fileScanner.Text()
		r, err := regexp.Compile(`mul\((\d+),(\d+)\)`)
		check(err)
		out := r.FindAllStringSubmatch(lineStr, -1)
		for _, match := range out {
			x, _ := strconv.Atoi(match[2])
			y, _ := strconv.Atoi(match[1])
			total += x * y
		}
	}
	fmt.Println("total: ", total)
	file.Close()
}

File: 2025/day-6/part1.py
file_name = "input.txt"
with open(file_name, "r") as f:
    lines = [x.strip().split() for x in f.read().strip().split("\n")]

overall = 0
for i in range(len(lines[0])):
    op = lines[-1][i]
    total = 1 if op == "*" else 0
    for j in range(len(lines) - 1):
        if op == "*":
            print(f"{int(lines[j][i])=}")
            total *= int(lines[j][i])
        else:
            total += int(lines[j][i])
    overall += total

    print(f"{total=}")
print(f"{overall=}")
